lambdas/credentialissuerreturn/src/main/java/uk/gov/di/ipv/core/credentialissuerreturn/CredentialIssuerReturnHandler.java

```diff
@@ -106,12 +112,28 @@ private Optional<ErrorResponse> validate(CredentialIssuerRequestDto request) {
             return Optional.of(ErrorResponse.MISSING_IPV_SESSION_ID);
         }
 
+        if (StringUtils.isBlank(request.getState())) {
+            return Optional.of(ErrorResponse.MISSING_OAUTH_STATE);
+        }
+
+        if (!request.getState().equals(getPersistedOauthState(request))) {
+            return Optional.of(ErrorResponse.INVALID_OAUTH_STATE);
+        }
+
         if (getCredentialIssuerConfig(request) == null) {
             return Optional.of(ErrorResponse.INVALID_CREDENTIAL_ISSUER_ID);
         }
         return Optional.empty();
     }
 
+    @Tracing
+    private String getPersistedOauthState(CredentialIssuerRequestDto request) {
```

Comment:
Same lambda permissions and env variables will be needed for the cri return lambda

Reply:
Done!